src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.OperationException;
+
+public class PurchaseOperation implements Operation {
+    private final StorageDao storageDao;
+
+    public PurchaseOperation(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void proceed(FruitTransaction fruitTransaction) {
+        if (storageDao.get(fruitTransaction.getFruitName()) == null) {
+            throw new OperationException("Fruit " + fruitTransaction.getFruitName()
+                    + " doesn't exist");
+        }
+        Fruit fruit = storageDao.get(fruitTransaction.getFruitName());
+        fruit.setQuantity(fruit.getQuantity() - fruitTransaction.getQuantity());
```

Comment:
Handling Purchase operation requires checking the result balance before saving it in the Storage - it should be positive. You need to throw a `RuntimeException` in case the balance is negative. This is to prevent the quantity of fruits from going below zero which would be an invalid state.